source/MaterialXGenShader/Syntax.cpp

```diff
@@ -112,11 +112,12 @@ string Syntax::getSwizzledVariable(const string& srcName, const TypeDesc* srcTyp
         }
         else
         {
-            if (it->second >= srcMembers.size())
+            int channelIndex;
+            if ((channelIndex = srcType->getChannelIndex(ch)) < 0)
             {
-                throw ExceptionShaderGenError("Given member in channels pattern is incorrect for type '" + srcType->getName() + "'.");
+                throw ExceptionShaderGenError("Given channel index: '" + string(1,ch) + "' in channels pattern is incorrect for type '" + srcType->getName() + "'.");
             }
-            membersSwizzled.push_back(srcName + srcMembers[it->second]);
+            membersSwizzled.push_back(srcName + srcMembers[channelIndex]);
```

Comment:
The check to make sure the index is not greater than srcMember size could still be done here before using it. Just in case there are some inconsistency between TypeSyntax srcMember length and the TypeDesc channel index returned. This check could be included as part of the check for negative index above.